FEAT: Race eat: Moth-Nian cloth, Vox Tech, Drask Soap

PhantornRU opened the pull request:
Продолжение функционала: https://github.com/ss220-space/Paradise/pull/1585
Моли оффов: https://github.com/ParadiseSS13/Paradise/pull/17000

Также преподготовка к введению будущих расовых уникальных диет, возвращение ЛОРной диенты ВОКСов и Драсков как это и положено. Для у ОФФов такого нету.

## What Does This PR Do
- Моли теперь могут есть одежду. 
- Воксам возвращена ЛОРная возможность пожирать мелких технологичных предметов.
- Драски по нормальному едят мыло.
- У предметов появился свой кодовый тип для обозначения на будущие расовые предметные диеты для тех же воксов и драсков.
- С помощью HARM интента, возможно насильное скармливание себе/другому, вплоть до того, что жертва почувствует слабость и в итоге блеванет.
- На насильное скармливание установлена задержка.
- Тем не менее, некоторые предметы с собственными функциями - возможно "съесть" только при GRAB интенте. Из-за чего не получится съесть анализатор здоровья "случайно" покормив себя при анализе. Или анализе другого вокса.
- Анимация "стукания предметом одежды" больше не проходит если необходимо жрац.
- Хайриски и крайне ценные вещи, которые невозможно получить никаким способом жрать нельзя.
- Есть можно до определенных порогов. После чего можно только насильно запихивать и заставлять страдать жертву или себя.
- Поеденный предмет меняет свой цвет из-за поврежденной целостности INTEGRITY
- В описании для "едока" пишется можно ли есть предмет. И как его есть, аккуратно GRAB'ом или HELP/GRAB интентом
- Всем пишется что это за предмет и для кого он подойдет ("Это тканевый предмет" - значит подойдет для молей.)
- Пишется насколько покусан предмет.

## Why It's Good For The Game
Доработанное дополнение к нашим диетам, которое ранее было плохо реализовано древними Парадайзами, а также не реализовано ОФФами.

## Images of changes
![image](https://user-images.githubusercontent.com/41479614/202873496-38bf0e15-def1-4691-a1d8-047af01a2ccf.png)
![image](https://user-images.githubusercontent.com/41479614/202873489-bb541b98-3861-4830-b5af-8a90f1337fc6.png)
![image](https://user-images.githubusercontent.com/41479614/202873504-d51cc245-899a-4d3b-9440-ebf1b4cf4e40.png)
![image](https://user-images.githubusercontent.com/41479614/202873553-2403abaa-1cd1-4136-87e3-7ad12be099be.png)
![image](https://user-images.githubusercontent.com/41479614/202873562-cfb86937-cdbb-42d6-9f43-a045c86350fd.png)
![image](https://user-images.githubusercontent.com/41479614/202890700-88a40a85-b400-4e7b-97c9-42a5b369a67c.png)
![image](https://user-images.githubusercontent.com/41479614/202891657-7a3e20f9-ee1b-408f-93f3-05a08bb0094b.png)
![image](https://user-images.githubusercontent.com/41479614/202891679-1e08b9ef-fd7c-4489-b18a-d78b216b7753.png)
![image](https://user-images.githubusercontent.com/41479614/202893356-315a6ed7-bc88-417b-b722-ed35ff3299df.png)
![image](https://user-images.githubusercontent.com/41479614/202893455-b217fbda-c3c4-44fb-8fab-b7ae8e5c95ab.png)
![image](https://user-images.githubusercontent.com/41479614/203836744-f4724897-3940-4983-ae18-268156b351d1.png)
![image](https://user-images.githubusercontent.com/41479614/203836863-27b3f50b-d3ff-43da-b7a9-d4610628dfd5.png)


## Станбатон и воксы
Единственное оружие которое можно съесть: Станбатон/Станпрод.
Его же можно скормить ВОКСам. Но только после ожидания прохождения КД.
Но съесть его можно только при GRAB интенте. И если он будет включен - он вас йобом токнет.
![image](https://user-images.githubusercontent.com/41479614/202892727-2c278b79-05c7-4334-b239-7eaa3a956b1b.png)

PhantornRU committed:
FEAT: Moth-Nian cloth eat

Всеравно этот Integrity нигде не используется.
FEAT: Moth-Nian cloth eat
Добавление задержи при кормежке
Рефактор, воксы могут жрать технологии
Расширения списка еды для воксов, молей и драсков
Нельзя жрать оружие, хайриски
Расширение списка поедания предметов
Расширение списка поедальных предметов
Update code/modules/food_and_drinks/item_food/eat_items.dm
Feat: add moth species
Изменение цвета при повреждениях

Co-Authored-By: Den109G <87372121+Den109G@users.noreply.github.com>

BeebBeebBoob commented:
Мдэ... Лучше бы этот ПР шёл на ПР ТМ молей а не отдельно + Азиз указывал в важных инфо что надо делать ПРы на мастер, не ТМ. Он сам потом решит на ТМ.

PhantornRU commented:
> Мдэ... Лучше бы этот ПР шёл на ПР ТМ молей а не отдельно + Азиз указывал в важных инфо что надо делать ПРы на мастер, не ТМ. Он сам потом решит на ТМ.
Да? Лять. А ведь действительно. Ну да ладно. Если шо потом на мастер перенесу.

BeebBeebBoob commented:
Вот куда стоит идти этот ПР https://github.com/MataVsn/Paradise/tree/mothsupremacy

Den109G commented on code/modules/food_and_drinks/item_food/eat_items.dm:
```diff
@@ -0,0 +1,93 @@
+		if(MATERIAL_CLASS_SOAP)
+			if(!isdrask(target))
+				return FALSE
+	if (is_only_grab_intent && user.a_intent != INTENT_GRAB)
```
```suggestion
	if(is_only_grab_intent && user.a_intent != INTENT_GRAB)
```
Если уж и делать, то держать один стиль.
А вообще, нам нужно это? Может оставить на один интент?

PhantornRU replied:
Да, нам нужно это. Дело в том что ты всегда обычную еду ешь на Help интенте как правило. Тебе не нужно в неё хвататься "грабом" чтобы поесть. Ибо в грабе тебя могут буквально толкнуть/подвинуть и т.п.
И разделение на ГРАБ позволит тебе не сидеть в нём постоянно и толкать народ, а использовать его только в случаях, когда предмет имеет несколько функционалов.
Если бы я мог, я бы оставил всё на хелп-интенте. Но из-за юзательности условных хилс-сканеров, хелп интент оставляется им, а уже граб - для поедания.

Den109G replied:
Может тогда только на граьб есть НЕ-еду?

PhantornRU replied:
Излишнее усложнение, как и написал выше. Лучше иметь 2 интента который в этом плане не взаимодействуют с куклой при держании вещей в руке.

PhantornRU commented:
Оно поломалось. Должно быть 600/9 meriCat

PhantornRU commented:
@Dimach ребейзнул под мейн. Повторно натестил. Можно залить на тестмердж под Клео и Вега?

PhantornRU committed:
Уменьшение значений получаемых нутриентов за одежду

Теперь нужны съесть 2 джампсьюта, вместо половинки джампсьюта чтобы насытиться.

PhantornRU committed:
Снижение получаемых нутриентов за технологии

PhantornRU committed:
Выпил прока, теперь идет запись прямо в файл
